src/v2/codegen/emitter.rs

```diff
@@ -211,6 +211,22 @@ pub trait Emitter: Sized {
         let state = self.state();
         state.reset_internal_fields();
 
+        let m = state.get_meta();
+        let borrowed_cm = m.borrow();
+
+        if borrowed_cm.is_none() {
+            if let Some(meta) = &mut *m.borrow_mut() {
```

Comment:
The first problem is that `borrowed_cm` is already borrowed immutably, so a mutable borrow will cause a panic at runtime (in this case, during tests). We can get rid of `borrowed_cm` entirely, since we're not using beyond it the `if` condition.

The other problem is that if we've already entered the block, it means that crate meta is `None`, so we won't ever satisfy the condition which is checking for `Some` pattern. What we should do is if it's `None`, then we should create a `CrateMeta`, set the name and version, and call `set_meta`.